kubernetes/controllers/resources/suite_test.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resources
+
+import (
+	"path/filepath"
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+	"sigs.k8s.io/controller-runtime/pkg/envtest/printer"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+
+	resourcesv2 "github.com/gravitational/teleport-plugins/kubernetes/apis/resources/v2"
+	resourcesv5 "github.com/gravitational/teleport-plugins/kubernetes/apis/resources/v5"
+	//+kubebuilder:scaffold:imports
+)
+
+// These tests use Ginkgo (BDD-style Go testing framework). Refer to
```

Comment:
Ginkgo is the default testing framework when you start the project using `operator-sdk`
I tried to stick as much to that base setup as possible

However, I was hesitant in using it because I don't think we use it in any other projects

I can change to testify, there aren't that many tests and the tests for User and Role have almost the same structure
So, right now, it should be a quick change
Do you think it's worth it?